### PR TITLE
feat: Use jupyter-book for documentation

### LINKED ISSUE #10: Link docstrings to the Jupyter Book
Link the docstrings to the Jupyter Book to ensure that documentation is all in one place and automatically updated when changes are made to the docstrings. Refer this PR for more information - https://github.com/uw-ssec/offshore-geodesy/pull/81

### LINKED ISSUE #31: Incompatibility between autoapi and autodoc_pydantic
This issue tracks a bug where I am facing issue with generating docs for python modules using PyDantic data models. I was able to find the root cause of this issue with @lsetiawan's help. I will need to replace `autoapi` with `autodoc` to resolve this issue.

### PR DESCRIPTION
**Pull request by aniketfadia96**
Hi, I have made the following changes in this PR -

1. Remove sphinx documentation builder
2. Use `jupyter-book` for generating documentation
3. Add `codespell` pre-commit hook to fix minor spelling mistakes 

Closes #10 
Closes #31

### CODE REVIEW

**Review by carlosgjs — APPROVED** [review]
LGTM, but it would. be great to get @lsetiawan to chime since he knows this space.

File: docs/requirements.txt [hunk 1]
```diff
@@ -1,9 +1,3 @@
...
-sphinx-autoapi==2.0.1
-nbsphinx
-ipython
-jupytext
-jupyter
+jupyter-book
 matplotlib
-numpy
\ No newline at end of file
+numpy
```
Comment (lsetiawan) [inline docs/requirements.txt:3]: If you already have your docs requirement in your `pyproject.toml`, you probably don't need this file anymore.

File: .github/workflows/build-documentation.yml [hunk 2]
```diff
@@ -30,6 +30,3 @@ jobs:
     - name: Install notebook requirements
       run: |
         sudo apt-get install pandoc
```
Comment (lsetiawan) [inline .github/workflows/build-documentation.yml:32]: If you set up read the docs correctly to build your docs then you shouldn't need this gh workflow anymore.

**Review by lsetiawan — CHANGES_REQUESTED** [review]
Could you please set up the read the docs pull request preview? https://docs.readthedocs.io/en/stable/guides/pull-requests.html

I'd like to see this in action ensuring the the configuration is correct and that the page is rendering. I see on the readthedocs end that you've set up the project, let me know if you wanna meet 1:1 to get this sorted. Thanks!

File: docs/requirements.txt (hunk 1 above)
Comment (aniketfadia96) [reply docs/requirements.txt:3]: Okay, thank you for this information. The jupyter-book default setup had this file, so I wasn't sure about it.

File: .github/workflows/build-documentation.yml (hunk 2 above)
Comment (aniketfadia96) [reply .github/workflows/build-documentation.yml:32]: Okay got it

**Comment by aniketfadia96** [comment]
> Could you please set up the read the docs pull request preview? https://docs.readthedocs.io/en/stable/guides/pull-requests.html
> 
> I'd like to see this in action ensuring the the configuration is correct and that the page is rendering. I see on the readthedocs end that you've set up the project, let me know if you wanna meet 1:1 to get this sorted. Thanks!

Okay that's a great suggestion. I have enabled by following the steps in the above link. Thank you!

**Comment by aniketfadia96** [comment]
I will go ahead and merge this PR to see the `jupyter-book` in action. Thank you @lsetiawan for your suggestions! 😄

**Comment by lsetiawan** [comment]
> I will go ahead and merge this PR to see the jupyter-book in action

FYI, You should have been able to activate and reopen PR to see it in action before merging it to ensure that the readthedocs jupyterbook workflow works before merging.

**Comment by lsetiawan** [comment]
I don't seem to see the updates on `https://neglected-diagnostics.readthedocs.io/en/latest/` the build might have failed.

**Comment by lsetiawan** [comment]
Could you give me owner permission to this readthedocs project?

**Pull request merged**